tests/test_downloader_handlers.py

```diff
@@ -107,7 +107,7 @@ def test_lazy_handlers(self):
 class FileTestCase(unittest.TestCase):
 
     def setUp(self):
-        self.tmpname = self.mktemp()
+        fd, self.tmpname = mkstemp()
```

Comment:
This actually creates the file and it's not deleted later. The old code only creates the file with the `^` suffix and deletes it in `tearDown`. As far as I can see the test just needs a temporary file with `^` as a suffix, so we can just use `suffix='^'` as proposed in a now hidden comment (not with a context manager though, as we don't want it to be deleted right away).

Reply:
Feel free to edit this PR if need be 👍

Reply:
#5736 

So using `suffix='^'` indeed should work.

Reply:
Done as part of the [merge with the main branch](https://github.com/scrapy/scrapy/pull/5285/commits/8cf1f9fb106a0b3cdee579d9b661035531ea7195).